homeassistant/components/light/tplink.py

```diff
@@ -196,6 +206,8 @@ def supported_features(self):
 
     def get_features(self):
         """Determine all supported features in one go."""
+        self._sysinfo = self.smartbulb.sys_info
```

Comment:
is this doing I/O ? And if it does, should we only fetch it if `self._sysinfo is None` ?

Reply:
Ah I see that it's only executing once.

I would suggest that you set `self._supported_features` to `None` to indicate it's not loaded and then initialize it to 0 in this method. It's now possible that you get a light with 0 features that will end up fetching features on each update.

Reply:
Done.